### ISSUE TITLE
🐛 Bug Report: Home Page can still be viewed, even if toggled to be hidden.

### ISSUE THREAD

**Issue by Hukuma1** [stonith404/pingvin-share#117]
### 👟 Reproduction steps

<img width="485" alt="hidden" src="https://user-images.githubusercontent.com/54511555/223176974-f35a84d8-29af-4618-a646-743e34e7feb5.png">

Using incognito mode (to simulate non-authenticated people viewing your page) if you refresh the main page and quickly click the Pingvin logo or server name at the top left, it will take you to the home page, even if you disabled it server-side. If you do this while signed in, you will see it flash.

<img width="456" alt="image" src="https://user-images.githubusercontent.com/54511555/223178443-530b4870-034c-407a-912c-d21ae6e8e8f4.png">

Not sure if related, but console also throws a few errors upon incognito visit:
```
_app-2d3d95bc4e5bfd44.js? Refresh token invalid or expired
/api/users/me:1          Failed to load resource: the server
```

### 👍 Expected behavior

Just show login screen, nothing else.

### 👎 Actual Behavior

Can be forced to show Home Screen even if it's turned off in settings.

### 🌐 Browser

Chrome

**Comment by stonith404** [comment]
Thanks for your report! I can't reproduce this maybe I do something wrong, could you share a short screen recording how you can trigger the issue?

**Comment by Hukuma1** [comment]
It's confusing because I just tried to re-create container from scratch and now I can't re-create it when logged in (the flashing part). But, I can still trigger it with an incognito tab.


https://user-images.githubusercontent.com/54511555/223531687-4cafbdc7-c9c2-47cc-96ec-ad6e20e2eade.mp4

This works if you refresh (F5) and then quickly click those top links. However, if you shift-F5 as the refresh, it will NOT work if you hit the links.

https://user-images.githubusercontent.com/54511555/223532713-05597dd1-a10c-4c33-a764-c1c3b2c3ccff.mp4

I am witnessing the same anomaly with the favicon. Here is me pressing F5 and I see default pingvin logo. If I shift-F5 it correctly pulls the custom logo (that beautiful purple box lol). Where is it finding the pingvin logo at all if all of my icons are the purple box?

`/Config/pingvin:/opt/app/frontend/public/img`
<img width="346" alt="image" src="https://user-images.githubusercontent.com/54511555/223533429-507d1719-4ca7-4aa7-b35f-0d72e9e9f06d.png">

<img width="569" alt="image" src="https://user-images.githubusercontent.com/54511555/223533537-c479649b-b57c-442f-a020-bc64cc915ab5.png">

I'm not sure if this is some cache issues with browsers, but I was able to reproduce the incognito tab in both Chrome and Edge. But regardless, it should not be finding the pingvin logo by default anywhere if all the logo files are replaced? (I think?)

**Comment by stonith404** [comment]
Thanks for your detailed information! I was able to reproduce the issue with the home page. It's strange because the issue doesn't occur in Firefox. I'll look into it.

Regarding the favicon issue, it looks like you did everything correctly. NextJS has built in image optimization that caches the images, maybe that's the issue. Can you try to run `docker compose down && docker compose up -d`, this deletes the cache. Please let me know if this worked so I can debug this further.

**Comment by Hukuma1** [comment]
I run this in a stack of a whole bunch of containers. Does taking down just the one container do the same thing? (if so, I've restarted it plenty). I even deleted the image and re-pulled it and still got the same results.

Not sure if it's viable, but it would be a lot easier if the user could supply the image e.g. 512x512 logo file and Pingvin automatically created all the files needed without users manually making the different sizes.

Could it be that my default images aren't crushed to some proper format maybe? Not sure. In case you want my logo files I'm testing this with: https://we.tl/t-V02zVaowNk

**Comment by stonith404** [comment]
Removing the container like you did should clear the cache. Maybe there is an issue with mounting the images with Docker, I have no clue... I've created an issue #120 for uploading the logo from the frontend and I'll work on it.

**Comment by stonith404** [comment]
Should be fixed in `v0.12.0`. Feel free to reopen the issue if you still encounter this issue.

**Comment by Hukuma1** [comment]
The flashing home screen is fixed on my end too. Thank you!

Didn't want to open new thread in case this was an easy fix, but it seems `icon-48x48.png` isn't being written over with a new logo.

<img width="515" alt="image" src="https://user-images.githubusercontent.com/54511555/224383276-9976d50f-a27d-4959-a4be-19c14306f5c2.png">

And perhaps that is why I'm seeing the 🐧 favicon still instead of the new logo. Not sure. There's definitely some caching going on inside somewhere as well.

https://user-images.githubusercontent.com/54511555/224393112-aa6336eb-cfed-4dbe-afe9-aacd912c25f9.mp4

**Comment by stonith404** [comment]
I've just fixed in 753dbe83b770814115a2576c7a50e1bac9dc8ce1 that the 48px icon gets updated correctly. This icon isn't used for the favicon only for the PWA icon. I think the favicon caching is set by the browser and I think it makes sense that it is cached as the favicon doesn't change that much. 

But the favicon is displayed, the problem is that it shows the old one, isn't it? As I said I can't really change the caching behaviour of the browser... `Ctrl` + `Shift` + `R` should clear the cache, does correct favicon get displayed then?

**Comment by Hukuma1** [comment]
So bizarre. Thanks for entertaining my comments. I ended up downloading both Firefox and Chrome portable and it seems to be an issue of my actual Chrome that's installed. Cleared cache, used dev tools/network page to clear it too. It just flashes back to the old image. But it doesn't matter, it's on my end only.

Thank you for adding the logo generator!

**Comment by Hukuma1** [comment]
Btw not sure how relevant it is, but probably safe to remove the `Sign Up` button at the top right if Home Page is hidden. It serves no purpose as the only thing you can do is sign in on this page since it doesn't even act as a submit button. And it'll probably remove the menu button on mobile since nothing to expand/collapse.

<img width="584" alt="image" src="https://user-images.githubusercontent.com/54511555/224491134-c50b2195-fb7b-4b3c-9cd4-25b3fba78998.png">

**Comment by stonith404** [comment]
Yeah that's a good point. If the home page and sign up is disabled, it doesn't make sense to have a single link in the navigation bar.
